src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -22,6 +24,12 @@ export const PartyMap: React.FC<PartyMapProps> = ({ venue }) => {
   const { user, profile } = useUser();
   const { recentVenueUsers } = useRecentVenueUsers();
 
+  const { relatedVenueIds } = useRelatedVenues({ currentVenueId: venue.id });
+
+  const venueIds = useMemo(() => relatedVenueIds, [relatedVenueIds]);
+
+  const { events: venueEvents } = useVenueEvents({ venueIds: venueIds });
```

Comment:
`relatedVenueEvents` would probably be a better name here.

Reply:
Though this is going to fetch venues for literally every related venue, parents, children, siblings, etc. Which I don't believe we need here at all.

All we need is the events for the venues of the rooms shown on this map, which I believe I described in my original overview/explanation (https://github.com/sparkletown/sparkle/issues/1487) of how to resolve this issue.

Reply:
I've now implemented a filter for venues that have `parentId` set to the current `venue.id` or the venue that is the current `venue`: 
 
`const relatedVenueRoomIds = relatedVenues.filter((roomVenue) => roomVenue.parentId === venue.id || roomVenue.id === venue.id).map((roomVenue) => roomVenue.id);`